middleware/src/main/java/com/nytimes/android/external/store/middleware/GsonParserFactory.java

```diff
@@ -0,0 +1,73 @@
+package com.nytimes.android.external.store.middleware;
+
+import com.google.gson.Gson;
+import com.nytimes.android.external.store.base.Parser;
+
+import java.io.Reader;
+
+import okio.BufferedSource;
+
+/**
+ * Factory which returns various Gson {@link Parser} implementations.
+ */
+
+public class GsonParserFactory {
+
+    /**
+     * Returns a new Parser which parses from {@link Reader} to the specified type, using
+     * the provided {@link Gson} instance.
+     */
+    public static <T> Parser<Reader,T> createReaderParser(Gson gson, Class<T> parsedClass) {
+        if (gson == null) throw new IllegalArgumentException("gson cannot be null.");
+        if (parsedClass == null) throw new IllegalArgumentException("parsedClass cannot be null.");
+        return new GsonReaderParser<>(gson,parsedClass);
+    }
+
+    /**
+     * Returns a new Parser which parses from {@link Reader} to the specified type, using
+     * a new default configured {@link Gson} instance.
+     */
+    public static <T> Parser<Reader, T> createReaderParser(Class<T> parsedClass) {
+        if (parsedClass == null) throw new IllegalArgumentException("parsedClass cannot be null.");
+        return new GsonReaderParser<>(new Gson(), parsedClass);
```

Comment:
The other trick is that the factory is all static and has no state. 
I'm not a fan of static variables that are mutable.